utils_nlp/models/transformers/extractive_summarization.py

```diff
@@ -0,0 +1,732 @@
+            warmup_steps (int, optional): Number of steps taken to increase learning rate from 0
+                to `learning_rate`. Defaults to 1e5.
+            learning_rate (float, optional):  Learning rate of the AdamW optimizer. Defaults to
+                5e-5.
+            optimization_method (string, optional): Optimization method used in fine tuning.
+            max_grad_norm (float, optional): Maximum gradient norm for gradient clipping.
+                Defaults to 0.
+            gradient_accumulation_steps (int, optional): Number of batches to accumulate
+                gradients on between each model parameter update. Defaults to 1.
+            decay_method (string, optional): learning rate decrease method. Default to 'noam'.
+            report_every (int, optional): The interval by steps to print out the trainint log.
+                Defaults to 50.
+            beta1 (float, optional): The exponential decay rate for the first moment estimates.
+                Defaults to 0.9.
+            beta2 (float, optional): The exponential decay rate for the second-moment estimates.
+                This value should be set close to 1.0 on problems with a sparse gradient.
+                Defaults to 0.99.
+            verbose (bool, optional): Whether to print out the training log. Defaults to True.
+            seed (int, optional): Random seed used to improve reproducibility. Defaults to None. 
+        """
+
+        device, num_gpus = get_device(num_gpus=num_gpus, local_rank=local_rank)
+
+        def move_batch_to_device(batch, device):
+            return batch.to(device)
+
+        # if isinstance(self.model, nn.DataParallel):
+        #    self.model.module.to(device)
+        # else:
+        self.model.to(device)
+
+        optimizer = model_builder.build_optim(
+            optimization_method,
+            learning_rate,
+            max_grad_norm,
+            beta1,
+            beta2,
+            decay_method,
+            warmup_steps,
+            self.model,
+            None,
+        )
+
+        # train_dataloader = get_dataloader(train_iter(), is_labeled=True, batch_size=batch_size)
+
+        super().fine_tune(
+            train_dataloader=train_dataloader,
+            get_inputs=ExtSumProcessor.get_inputs,
+            device=device,
+            move_batch_to_device=move_batch_to_device,
+            n_gpu=num_gpus,
+            num_train_epochs=-1,
+            max_steps=max_steps,
+            optimizer=optimizer,
+            warmup_steps=warmup_steps,
+            gradient_accumulation_steps=gradient_accumulation_steps,
+            verbose=verbose,
+            seed=seed,
+            report_every=report_every,
+            clip_grad_norm=False,
+            max_grad_norm=max_grad_norm,
+        )
+
+    def predict(
+        self,
+        test_dataset,
+        num_gpus=1,
+        batch_size=16,
+        sentence_seperator="<q>",
```

Comment:
This argument doesn't seem to be used. Should it be passed to get_pred? 
Also typo in "seperator".